### PR TITLE
Add recipe configs validating

### PR DESCRIPTION
**Pull request by BoxiangW**
# What does this PR do ?

Add model config validations for NeMo 2.0

**Collection**: llm

# Changelog 
- Added training and fine-tuning config validation basics for NeMo 2.0.
- Added unit tests for config validation. 

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by dimapihtar** [comment]
could you also update your branch to latest main, please? I think Pablo have made auto conf tests not optional recently.

File: nemo/collections/llm/api.py [hunk 1]
```diff
@@ -716,3 +719,99 @@ def _set_with_io(obj, attr, value):
...
+    ), "Global batch size must be divisible by micro batch size in data module."
+
+    ## Trainer validation
+
+    # Basic validation
+    assert trainer.strategy.tensor_model_parallel_size > 0
+    assert trainer.strategy.pipeline_model_parallel_size > 0
+    assert trainer.strategy.context_parallel_size > 0
+
+    # DP validation
```
Comment (marcromeyn) [inline nemo/collections/llm/api.py:763]: You are assuming here that a user always uses `MegatronStrategy`, that's not the case. So please add some if-statements in this function

**Comment by marcromeyn** [comment]
@maanug-nv How does this relate to your work on validation of configs?

File: nemo/collections/llm/api.py (hunk 1 above)
Comment (BoxiangW) [reply nemo/collections/llm/api.py]: Thanks for pointing out. I have fixed this part.

**Comment by BoxiangW** [comment]
Hi @marcromeyn , can you help take a look at this PR again? Thanks

**Review by marcromeyn — APPROVED** [review]

**Pull request merged**